src/app/handlers/new_member.handler.ts

```diff
@@ -0,0 +1,38 @@
+import { GuildMember, Role } from 'discord.js';
+import { IContainer, IHandler } from '../../common/types';
+import { Maybe } from '../../common/types';
+
+export class NewMemberHandler implements IHandler {
+  private _roleCache: Record<string, Maybe<Role>> = {
+    'Un Acknowledged': undefined,
```

Comment:
personal preference, i would create something like `const UNACKNOWLEDGE_ROLE = 'Un Acknowledged';` since you reference this string 4 times. if you ever have to change what this string is, it would only be in one place you need to change instead of 4 places.

Reply:
When i do this, the type of the role does not turn into a `Role` rather than `Maybe<Role>` after the check for undefined.
With this code
```
if (!this._roleCache[this._UNACKNOWLEDGED_ROLE]) {
      return;
}
const role = this._roleCache[this._UNACKNOWLEDGED_ROLE];
```
role is still type `Maybe<Role>`
![image](https://user-images.githubusercontent.com/39809146/98746631-df5c7b00-2383-11eb-9ec7-c2c0a49d419e.png)

Reply:
`this._UNACKNOWLEDGED_ROLE` is a string

Reply:
Actually, I could just assert it as a role, since I know it's not undefined after the if

Reply:
yeah i would just tell typescript it's a role

`const role = this._roleCache[this._UNACKNOWLEDGED_ROLE] as Role`

Reply:
The `const role = ...` was redundant,I just did that to show it's type. 
I asserted in inline `member.addRole(<Role>this._roleCache[this._UNACKNOWLEDGED_ROLE]);`